src/wiki_ulmfit.py

```diff
@@ -0,0 +1,99 @@
+import logging
+
+import gdown
+import sentencepiece as spm
+from fastai import *
+from fastai.text import *
+from gensim.corpora import WikiCorpus
+
+from src.utils import LangTokenizer
+
+logger = logging.getLogger()
+
+
+class WikiTrainer:
+    def __init__(self, path, lang="id"):
+        self.path = path
+        self.lang = lang
+
+    def download_wiki_files(self, wiki_text=False):
+        """
+        Downloads Processed Wikipedia from google drive. 
+        """
+        lang_dict = {
+            "id": {
+                "wiki": "https://drive.google.com/uc?export=download&id=10P95rNlMPuHyB57k40KFcZUlaXdFopUC",
+                "data": "https://drive.google.com/uc?export=download&id=1-0gaC0bmYyMoUWIkGoH6fY6DUQTArMD3",
+            }
+        }
+        if not (self.path / "data_save.pkl").exists():
+            gdown.download(
+                lang_dict[self.lang]["data"], f"{self.path}/data_save.pkl", quiet=False
+            )
+        if wiki_text:
+            gdown.download(
+                lang_dict[self.lang["wiki"]],
+                f"{self.path}/{self.lang}_wiki.txt",
+                quiet=False,
+            )
+
+    def load_data_lm(self):
+        """
+        Loads data_lm for wikipedia data.
+        """
+        tokenizer = Tokenizer(tok_func=LangTokenizer, lang=self.lang)
+        if (self.path / "data_save.pkl").exists():
+            data_lm = load_data(self.path)
+        else:
+            data_lm = TextLMDataBunch.from_csv(
+                self.path,
+                f"{self.lang}_wiki.txt",
```

Comment:
Paramterize the filename as an arg of the function or ideally, the class init. This way, you can do the `exists()` check for all necessary ingredients in one place. 

And not waste time down the pipeline if something is broken.